landoapi/api/uplift.py

```diff
@@ -8,68 +8,140 @@
+    repos = [
+        phab.expect(repo, "fields", "name") for repo in get_uplift_repositories(phab)
+    ]
+
+    return {"repos": repos}, 201
+
+
 @require_phabricator_api_key(optional=False)
 @auth.require_auth0(scopes=("lando", "profile", "email"), userinfo=True)
 def create(data):
     """Create new uplift requests for requested repository & revision"""
+    repo_name = data["repository"]
+    landing_path = parse_landing_path(data["landing_path"])
+    tip_revision_id = landing_path[-1][0]
+    phab: PhabricatorClient = g.phabricator
 
-    # Validate repository
+    # Validate repository.
     all_repos = get_repos_for_env(current_app.config.get("ENVIRONMENT"))
-    repository = all_repos.get(data["repository"])
-    if repository is None or not repository.approval_required:
+    repository = all_repos.get(repo_name)
+    if repository is None:
         return problem(
             400,
-            "No valid uplift repository",
-            "Please select an uplift repository to create that uplift request.",
+            f"Repository {repo_name} is not a repository known to Lando.",
+            "Please select an uplift repository to create the uplift request.",
             type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
         )
 
-    logger.info(
-        "Checking approval state",
-        extra={
-            "revision": data["revision_id"],
-            "target_repository": data["repository"],
-        },
-    )
-    is_approval, revision, target_repository = check_approval_state(
-        g.phabricator,
-        revision_id=data["revision_id"],
-        target_repository_name=data["repository"],
-    )
-    logger.info(
-        "Approval state is valid",
-        extra={"style": "approval" if is_approval else "uplift"},
-    )
+    if not repository.approval_required:
+        return problem(
+            400,
+            f"Repository {repo_name} is not an uplift repository.",
+            "Please select an uplift repository to create the uplift request.",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
+        )
 
     try:
-        if is_approval:
-            output = create_approval_request(
-                g.phabricator, revision, data["form_content"]
-            )
-        else:
-            output = create_uplift_revision(
-                g.phabricator, revision, target_repository, data["form_content"]
-            )
-    except Exception as e:
-        logger.error(
-            "Failed to create an uplift request",
+        logger.info(
+            "Checking approval state",
             extra={
-                "revision": data["revision_id"],
-                "repository": repository,
-                "error": str(e),
+                "revision": tip_revision_id,
+                "target_repository": repo_name,
             },
         )
-        raise
+        revision_data, target_repository = get_uplift_conduit_state(
```

Comment:
I think `get_uplift_conduit_state` could be doing a little more when it comes to the `target_repo`. For example, it could do some of the work being done in `create_uplift_revision` so that we can fail sooner if there's something wrong with the repo (and so that we don't have to pass the dictionary around since we're only using it for this one purpose.)

Alternatively, we could also parse the target repo after this call, but before `create_uplift_revision`.